packages/client/src/lib.js

```diff
@@ -94,15 +108,48 @@ class NFTStorage {
   }
   /**
    * @param {API.Service} service
-   * @param {Blob} blob
-   * @returns {Promise<API.CIDString>}
+   * @param {Blob|CarReader} carReader
+   * @param {{onStoredChunk?: (cid: API.CIDString) => void}} [options]
+   * @returns {Promise<API.CIDString | undefined>}
    */
-  static async storeCar({ endpoint, token }, blob) {
-    const car =
-      blob.type !== 'application/car'
-        ? blob.slice(0, blob.size, 'application/car') // TODO: test this in safari.
-        : blob
-    return NFTStorage.storeBlob({ endpoint, token }, car)
+  static async storeCar(
+    { endpoint, token },
+    carReader,
```

Comment:
Since it can be a `Blob` **or** a `CarReader`...

```suggestion
    car,
```